GTViewController/Classes/Custom Classes/GTFollowupModalView.m

```diff
@@ -0,0 +1,183 @@
+            currentY = inputFieldView.frame.origin.y + inputFieldView.frame.size.height + betweenElementsSpace;
+            
+            [self addSubview:inputFieldView];
+        } else if ([modalComponentElementName isEqual:kName_Button_Pair]) {
+            TBXMLElement *firstButtonElement = modalComponentElement->firstChild;
+            TBXMLElement *secondButtonElement = firstButtonElement->nextSibling;
+            
+            GTMultiButtonResponseView *buttonPairView = [[GTMultiButtonResponseView alloc] initWithFirstElement:firstButtonElement
+                                                                                                  secondElement:secondButtonElement
+                                                                                                  parentElement:modalComponentElement
+                                                                                                      yPosition:currentY
+                                                                                                  containerView:self
+                                                                                                      withStyle:style];
+            
+            [self addSubview:buttonPairView];
+            
+            currentY += buttonPairView.frame.size.height + betweenElementsSpace;
+            
+        } else if ([modalComponentElementName isEqual:kName_Thank_You]) {
+            
+            self.thankYouView = [[GTFollowupThankYouView alloc]initFromElement:modalComponentElement
+                                                                     withFrame:self.frame
+                                                                 withPageStyle:style];
+            
+            NSArray *listeners = [[TBXML valueOfAttributeNamed:kAttr_listeners forElement:modalComponentElement] componentsSeparatedByString:@","];
+            
+            for (id listener in listeners) {
+                NSString *listenerName = listener;
+                
+                if ([interpreterDelegate respondsToSelector:@selector(registerListenerWithEventName:target:selector:parameter:)]) {
+                    [interpreterDelegate registerListenerWithEventName:listenerName
+                                                                target:interpreterDelegate
+                                                              selector:@selector(transitionFollowupToThankYou)
+                                                             parameter:nil];
+                    
+                }
+            }
+        }
+        
+        modalComponentElement = modalComponentElement->nextSibling;
+    }
+    
+    return self;
+}
+
+
+- (void)setWatermark:(UIImageView *) watermarkImageView {
+
+    [self addSubview:[self copyWatermark:watermarkImageView]];
+    [self.thankYouView addSubview:[self copyWatermark:watermarkImageView]];
+}
+
+
+/* see: http://stackoverflow.com/questions/25439234/how-to-clone-a-uiimageview */
```

Comment:
good catch, i didn't know that about uiimageview. If you have time I would make a category for uiimageview with a copy method that does this logic instead of putting it on this method. It will be more reusable.